ocaml/xapi/xapi_pool.ml

```diff
@@ -1646,6 +1646,15 @@ let sync_database ~__context =
 
 (* This also means me, since call will have been forwarded from the current master *)
 let designate_new_master ~__context ~host =
+  if Xapi_pool_helpers.pool_secret_rotation_pending ~__context then
```

Comment:
Should you also block pool secret rotation when designate new master is pending? Usually this is handled by using allowed operations.
Perhaps all these designate new master, and PSR should gain an operation type in xapi_pool_helpers, and then you can easily declare in one place that they are mutually exclusive.

Reply:
Also I assume you already block pool secret rotation when HA is enabled? (master transitions can happen unpredictably and they cannot be blocked by design: if old master dies then another pool member must be promoted to master)

Reply:
We discussed adding a PSR operation to xapi pool helpers in this PR (https://github.com/xapi-project/xen-api/pull/4162). We elected to use a flag in the database instead, because a PSR can be pending in between calls to Pool.secret_rotate (which didn't fit nicely with the semantics of current operations - in particular you can't wrap `Xapi_psr.start` inside `with_pool_operation` and expect it to work).

I'll add designate_new_master to xapi_pool_helpers, that's a good idea.

(and yes, HA and PSR are already mutually exclusive)